routes/tasks.ts

```diff
@@ -1,57 +1,107 @@
-import express, {Request, Response, NextFunction} from "express";
-import {Task} from "../models/task";
+import express, { Response, NextFunction } from "express";
+import { Task } from "../models/task";
+import jwt from "jsonwebtoken";
+import ICustomRequest from "../models/custom-request";
+import { checkToken } from "../middleware/checkToken";
 
 const router = express.Router();
 
-router.get('/', async (_req: Request, res: Response, next: NextFunction): Promise<void> => {
-    try {
-        const results = await Task.find({}).limit(10);
-        res.status(200).send(results);
-    } catch (e) {
-        console.error('Error fetching data:', e);
-        res.send('Server Error').status(500);
+router.get('/', checkToken, async (_req: ICustomRequest, res: Response, next: NextFunction): Promise<void> => {
+    if (_req.token) {
+        jwt.verify(_req.token, process.env.SECRET_KEY as string, async (err, authorizedData) => {
```

Comment:
instead of rewriting verification for the token, create a separate method where you pass the necessary fields, to make this check